pymeasure/instruments/keithley/keithley2602B.py

```diff
@@ -50,21 +55,45 @@ class Keithley2602B(Keithley2600):
             f"smu{channel}.trigger.IDLE_EVENT_ID",
         ]
 
-    def __init__(self, adapter, **kwargs):
-        super().__init__(adapter, includeSCPI=False, **kwargs)
+    def __init__(self, hostname: str, port: int = 5025, **kwargs) -> None:
+        """Initializes LAN communication with the device.
+        Flushes the error queue and logs the system identification info.
+        If ID info cannot be retrieved, or communication was not successfully
+        initialized, the results are logged.
 
+        Args:
+            hostname (str): Hostname of the system (initial expected value is a concatenation of
+                product number and serial number. Note that the hostname can be overwritten.
+            port (int): Port used to communicate with the system
+        """
+        self.hostname = hostname
+        self.port = port
+        adapter = f"TCPIP::{self.hostname}::{self.port}::SOCKET"
         self.dio_pins = [
             Keithley2600DigitalIOPin(self, i + 1) for i in range(self.number_of_pins)
         ]
 
+        try:
+            super().__init__(adapter, **kwargs)
+
+        except ValueError or socket_error:
+            log.exception(f"Could not connect to {self.hostname}", exc_info=True)
+            super().communication_success = False
+
+        # Override valid voltage ranges (as requested)
+        # to avoid potential damage to diodes
+        # This does not work as expected
+        # for channel_name in [self.ChA, self.ChB]:
+        #   channel_name.source_voltage_values = [-0.7, 0.7]
+        #   channel_name.compliance_voltage_values = [-0.7, 0.7]
```

Comment:
@sam0085 asked to restrict the valid voltage range further (so as to not damage any diodes). I tried to make use of the [PyMeasure dynamic properties](https://docs.python.org/3/library/abc.html#abc.abstractproperty) here, but the above does not work (i.e. the valid range is not overwritten dynamically - the valid range used for validation is the range defined at instantiation [here](https://github.com/XanaduAI/pymeasure/pull/2/files#diff-9e56db960c3230a49c224444f8f63e96e83c20966a70a9f26de5feaaa5f69095L208-R212)).

Reply:
This seems too useful to give up on...

Reply:
I fixed the issue - the `Channel` class doesn't inherit from `Instrument`, so it is missing the logic/methods needed to create the `Dynamic` properties. I made the `Channel` class inherit the necessary methods/attributes from the `Instrument` class by creating a `BaseChannel` class that `Channel` inherits from:
```
class BaseChannel(Instrument):
    def __init__(self):
        self._special_names = self._setup_special_names()
```
I don't think this is a "full-proof" solution, but for now it works - this is a known bug in PyMeasure, so it should be addressed eventually: https://github.com/pymeasure/pymeasure/issues/616

Reply:
Nice :detective: work!